HHS/Head-Start-TTADP
# [TTAHUB-1942] Frontend for goal nudge

## Description of change
Add goal nudge options to the RTR when creating a new goal

## How to test
Create a new goal in the RTR. The following should be true:
- You should see existing goals with similar texts presented as options. Note the API doesn't get queried until 15 characters have been typed
- Clicking on an existing goal moves you over to that page to edit it
- If the goal status for an identically named goal differs across grants it isn't shown
- FEI goals are included as options in the dropdown
- You can check "use OHS initiative goals" if all selected grant doesn't have one of the curated goals
- If you select a suspended goal, you see a modal which offers you the chance to back out or reopen the suspended goal
- You can now set the goal source on a per-grant level on the RTR

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1942


## Checklists

### Every PR



### Before merge to main


### Production Deploy


### After merge/deploy
